src/client/mixins/custom-object-aware.ts

```diff
@@ -0,0 +1,58 @@
+import * as Marketo from 'node-marketo-rest';
+export class CustomObjectAwareMixin {
+  client: Marketo;
+
+  public async createOrUpdateCustomObject(customObjectName, customObject: Record<string, any>) {
+    return this.client._connection.postJson(
+      `/v1/customobjects/${customObjectName}.json`,
+      {
+        action: 'createOrUpdate',
+        dedupeBy: 'dedupeFields',
+        input: [customObject],
+      },
+      {
+        query: {
+          _method: 'POST',
+        },
+      },
+    );
+  }
+
+  public async getCustomObject(customObjectName) {
+    return this.client._connection.get(
+      `/v1/customobjects/${customObjectName}/describe.json`,
+      { query: { _method: 'GET' } },
+    );
+  }
+
+  public async queryCustomObject(customObjectName, idField, filterType, filterValue) {
```

Comment:
- [ ] Suspect we'll want to make this flexible to support the associated `check` step, but also to make it compatible with objects with compound keys.

Seems like the args would need to be:
- `customObjectName` - a string, like it already is here
- `searchFields` - an object where the keys are dedupe or link field names (and their corresponding values)
- `requestedFields` - an array of field names to be retrieved

And then this method would perform the logic to determine which of the two query requests to be made:
- If `searchFields` is an object with a single key, perform pure `GET` with `filterType` set to that key.
- If `searchFields` has more than one key, perform the `POST` with `_method=GET` like it is here.

- [ ] Recommend renaming `idField` to `requestFields` or similar, and have it be an array of field IDs.